Spek

## Requesting new package

*Not every package request can be fulfilled. Our repository is big enough to imply some issues with maintaining and hosting.*

- [x] I have read [Termux policy on package requests](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#a-note-about-package-requests).

***

### Package description
<!-- REQUIRED -->

**Link to home page and sources**
<!-- REQUIRED -->
1. Home page:https://github.com/alexkay/spek
2. Source code:https://github.com/alexkay/spek

**Additional information**
Have you compiled or tried to compile the package on device?
* If it did not work then please provide the error you ran into.
* If it did work then please share build steps and any patches that you had to apply.


Dependencies: gettext, wxwidgets, catch, intltool (Perl: XML::PARSER)

Tried steps: 
1) Compiled and installed intltool, catch2, gettext, XML::PARSER
2)Compiled and installed wxwidgets using:
```
git clone --recursive https://github.com/wxWidgets/wxWidgets
cd wxWidgets
mkdir buildgtk
cd buildgtk
../configure --with-gtk=2 --enable-cmdline --enable-monolithic --with-expat=builtin --enable-stl --with-sdl --enable-utf8 --enable-utf8only --disable-unsafe_conv_in_wxstring --enable-gui --disable-xlocale --disable-gui
make -j8 2>&1 | tee log
```
cat log
``` 
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps gcc -c -o wxregex_regcomp.o -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -DNDEBUG -D__WX__  -pthread -Wall -Wundef -O2 -D_GNU_SOURCE  -fvisibility=hidden -fPIC -DPIC   ../src/regex/regcomp.c
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps gcc -c -o wxregex_regexec.o -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -DNDEBUG -D__WX__  -pthread -Wall -Wundef -O2 -D_GNU_SOURCE  -fvisibility=hidden -fPIC -DPIC   ../src/regex/regexec.c
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps gcc -c -o wxregex_regerror.o -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -DNDEBUG -D__WX__  -pthread -Wall -Wundef -O2 -D_GNU_SOURCE  -fvisibility=hidden -fPIC -DPIC   ../src/regex/regerror.c
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps gcc -c -o wxregex_regfree.o -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -DNDEBUG -D__WX__  -pthread -Wall -Wundef -O2 -D_GNU_SOURCE  -fvisibility=hidden -fPIC -DPIC   ../src/regex/regfree.c
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps gcc -c -o wxexpat_xmlparse.o -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -DNDEBUG -I./src/expat/expat -DHAVE_EXPAT_CONFIG_H -pthread -Wall -Wundef -O2 -D_GNU_SOURCE  -fvisibility=hidden -fPIC -DPIC    ../src/expat/expat/lib/xmlparse.c
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps gcc -c -o wxexpat_xmlrole.o -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -DNDEBUG -I./src/expat/expat -DHAVE_EXPAT_CONFIG_H -pthread -Wall -Wundef -O2 -D_GNU_SOURCE  -fvisibility=hidden -fPIC -DPIC    ../src/expat/expat/lib/xmlrole.c
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps gcc -c -o wxexpat_xmltok.o -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -DNDEBUG -I./src/expat/expat -DHAVE_EXPAT_CONFIG_H -pthread -Wall -Wundef -O2 -D_GNU_SOURCE  -fvisibility=hidden -fPIC -DPIC    ../src/expat/expat/lib/xmltok.c
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps g++ -c -o monodll_any.o       -I../src/regex -I../src/expat/expat/lib -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -D__WX__      -DWXBUILDING -I../src/stc/scintilla/include -I../src/stc/scintilla/lexlib -I../src/stc/scintilla/src -D__WX__ -DSCI_LEXER -DNO_CXX11_REGEX -DLINK_LEXERS -DwxUSE_BASE=1 -DWXMAKINGDLL -fPIC -DPIC -pthread -Wall -Wundef -Wunused-parameter -Wno-ctor-dtor-privacy -Woverloaded-virtual -O2 -I/data/data/com.termux/files/usr/include  -fvisibility=hidden -fvisibility-inlines-hidden   ../src/common/any.cpp
In file included from ../src/regex/regexec.c:32:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:830:9: warning: 'HAVE_GCC_ATOMIC_BUILTINS' macro redefined [-Wmacro-redefined]
#define HAVE_GCC_ATOMIC_BUILTINS 1
        ^
../include/wx/android/config_android.h:27:9: note: previous definition is here
#define HAVE_GCC_ATOMIC_BUILTINS
        ^
In file included from ../src/regex/regexec.c:32:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:978:9: warning: 'HAVE_NANOSLEEP' macro redefined [-Wmacro-redefined]
#define HAVE_NANOSLEEP 1
        ^
../include/wx/android/config_android.h:25:9: note: previous definition is here
#define HAVE_NANOSLEEP
        ^
In file included from ../src/regex/regexec.c:32:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1080:9: warning: 'SIZEOF_VOID_P' macro redefined [-Wmacro-redefined]
#define SIZEOF_VOID_P 8
        ^
../include/wx/android/config_android.h:46:9: note: previous definition is here
#define SIZEOF_VOID_P 4
        ^
In file included from ../src/regex/regexec.c:32:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1083:9: warning: 'SIZEOF_LONG' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 8
        ^
../include/wx/android/config_android.h:43:9: note: previous definition is here
#define SIZEOF_LONG 4
        ^
In file included from ../src/regex/regexec.c:32:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1092:9: warning: 'SIZEOF_SIZE_T' macro redefined [-Wmacro-redefined]
#define SIZEOF_SIZE_T 8
        ^
../include/wx/android/config_android.h:45:9: note: previous definition is here
#define SIZEOF_SIZE_T 4
        ^
In file included from ../src/common/any.cpp:12:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:830:9: warning: 'HAVE_GCC_ATOMIC_BUILTINS' macro redefined [-Wmacro-redefined]
#define HAVE_GCC_ATOMIC_BUILTINS 1
        ^
../include/wx/android/config_android.h:27:9: note: previous definition is here
#define HAVE_GCC_ATOMIC_BUILTINS
        ^
In file included from ../src/common/any.cpp:12:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:978:9: warning: 'HAVE_NANOSLEEP' macro redefined [-Wmacro-redefined]
#define HAVE_NANOSLEEP 1
        ^
../include/wx/android/config_android.h:25:9: note: previous definition is here
#define HAVE_NANOSLEEP
        ^
In file included from ../src/common/any.cpp:12:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1080:9: warning: 'SIZEOF_VOID_P' macro redefined [-Wmacro-redefined]
#define SIZEOF_VOID_P 8
        ^
../include/wx/android/config_android.h:46:9: note: previous definition is here
#define SIZEOF_VOID_P 4
        ^
In file included from ../src/common/any.cpp:12:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1083:9: warning: 'SIZEOF_LONG' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 8
        ^
../include/wx/android/config_android.h:43:9: note: previous definition is here
#define SIZEOF_LONG 4
        ^
In file included from ../src/common/any.cpp:12:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1092:9: warning: 'SIZEOF_SIZE_T' macro redefined [-Wmacro-redefined]
#define SIZEOF_SIZE_T 8
        ^
../include/wx/android/config_android.h:45:9: note: previous definition is here
#define SIZEOF_SIZE_T 4
        ^
In file included from ../src/regex/regcomp.c:33:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:830:9: warning: 'HAVE_GCC_ATOMIC_BUILTINS' macro redefined [-Wmacro-redefined]
#define HAVE_GCC_ATOMIC_BUILTINS 1
        ^
../include/wx/android/config_android.h:27:9: note: previous definition is here
#define HAVE_GCC_ATOMIC_BUILTINS
        ^
In file included from ../src/regex/regcomp.c:33:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:978:9: warning: 'HAVE_NANOSLEEP' macro redefined [-Wmacro-redefined]
#define HAVE_NANOSLEEP 1
        ^
../include/wx/android/config_android.h:25:9: note: previous definition is here
#define HAVE_NANOSLEEP
        ^
In file included from ../src/regex/regcomp.c:33:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1080:9: warning: 'SIZEOF_VOID_P' macro redefined [-Wmacro-redefined]
#define SIZEOF_VOID_P 8
        ^
../include/wx/android/config_android.h:46:9: note: previous definition is here
#define SIZEOF_VOID_P 4
        ^
In file included from ../src/regex/regcomp.c:33:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1083:9: warning: 'SIZEOF_LONG' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 8
        ^
../include/wx/android/config_android.h:43:9: note: previous definition is here
#define SIZEOF_LONG 4
        ^
In file included from ../src/regex/regcomp.c:33:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1092:9: warning: 'SIZEOF_SIZE_T' macro redefined [-Wmacro-redefined]
#define SIZEOF_SIZE_T 8
        ^
../include/wx/android/config_android.h:45:9: note: previous definition is here
#define SIZEOF_SIZE_T 4
        ^
In file included from ../src/regex/regfree.c:39:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:830:9: warning: 'HAVE_GCC_ATOMIC_BUILTINS' macro redefined [-Wmacro-redefined]
#define HAVE_GCC_ATOMIC_BUILTINS 1
        ^
../include/wx/android/config_android.h:27:9: note: previous definition is here
#define HAVE_GCC_ATOMIC_BUILTINS
        ^
In file included from ../src/regex/regfree.c:39:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:978:9: warning: 'HAVE_NANOSLEEP' macro redefined [-Wmacro-redefined]
#define HAVE_NANOSLEEP 1
        ^
../include/wx/android/config_android.h:25:9: note: previous definition is here
#define HAVE_NANOSLEEP
        ^
In file included from ../src/regex/regfree.c:39:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1080:9: warning: 'SIZEOF_VOID_P' macro redefined [-Wmacro-redefined]
#define SIZEOF_VOID_P 8
        ^
../include/wx/android/config_android.h:46:9: note: previous definition is here
#define SIZEOF_VOID_P 4
        ^
In file included from ../src/regex/regerror.c:32:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:830:9: warning: 'HAVE_GCC_ATOMIC_BUILTINS' macro redefined [-Wmacro-redefined]
In file included from ../src/regex/regfree.c:39:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h#define HAVE_GCC_ATOMIC_BUILTINS 1:
39        ^
:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
../include/wx/android/config_android.h:/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h27::10839: note: previous definition is here
#define HAVE_GCC_ATOMIC_BUILTINS
        ^
:9: warning: 'SIZEOF_LONG' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 8
        ^
../include/wx/android/config_android.h:In file included from 43../src/regex/regerror.c::32:
9In file included from :../src/regex/regguts.h: 38:
noteIn file included from : ../src/regex/regcustom.h:39previous definition is here:

In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:978:9: warning: 'HAVE_NANOSLEEP' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 4
        ^
#define HAVE_NANOSLEEP 1
        ^
../include/wx/android/config_android.h:25:9: note: previous definition is here
#define HAVE_NANOSLEEP
        ^
In file included from In file included from ../src/regex/regerror.c:../src/regex/regfree.c32:
:In file included from ../src/regex/regguts.h:3938:
:
In file included from ../src/regex/regcustom.hIn file included from :39../src/regex/regguts.h:
In file included from :../include/wx/defs.h:3845:
:
In file included from In file included from ../include/wx/platform.h:../src/regex/regcustom.h159::
39/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:
:In file included from 1080:../include/wx/defs.h9:: 45warning:
: In file included from 'SIZEOF_VOID_P' macro redefined [-Wmacro-redefined]
../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1092:9: warning: 'SIZEOF_SIZE_T' macro redefined [-Wmacro-redefined]#define SIZEOF_VOID_P 8

        ^
../include/wx/android/config_android.h:46:9: note: previous definition is here
#define SIZEOF_VOID_P 4
        ^
#define SIZEOF_SIZE_T 8
        ^
In file included from ../src/regex/regerror.c:32:
In file included from ../src/regex/regguts.h:38:
../include/wx/android/config_android.hIn file included from ../src/regex/regcustom.h::3945:
In file included from :../include/wx/defs.h9:45::
 In file included from ../include/wx/platform.hnote:159: :
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.hprevious definition is here:
1083:9: warning: 'SIZEOF_LONG' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 8
        ^
../include/wx/android/config_android.h:43:9: note: previous definition is here#define SIZEOF_SIZE_T 4

        ^
#define SIZEOF_LONG 4
        ^
In file included from ../src/regex/regerror.c:32:
In file included from ../src/regex/regguts.h:38:
In file included from ../src/regex/regcustom.h:39:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1092:9: warning: 'SIZEOF_SIZE_T' macro redefined [-Wmacro-redefined]
#define SIZEOF_SIZE_T 8
        ^
../include/wx/android/config_android.h:45:9: note: previous definition is here
#define SIZEOF_SIZE_T 4
        ^
5 warnings generated.
5 warnings generated.
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps g++ -c -o monodll_appbase.o       -I../src/regex -I../src/expat/expat/lib -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -D__WX__      -DWXBUILDING -I../src/stc/scintilla/include -I../src/stc/scintilla/lexlib -I../src/stc/scintilla/src -D__WX__ -DSCI_LEXER -DNO_CXX11_REGEX -DLINK_LEXERS -DwxUSE_BASE=1 -DWXMAKINGDLL -fPIC -DPIC -pthread -Wall -Wundef -Wunused-parameter -Wno-ctor-dtor-privacy -Woverloaded-virtual -O2 -I/data/data/com.termux/files/usr/include  -fvisibility=hidden -fvisibility-inlines-hidden   ../src/common/appbase.cpp
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps g++ -c -o monodll_arcall.o       -I../src/regex -I../src/expat/expat/lib -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -D__WX__      -DWXBUILDING -I../src/stc/scintilla/include -I../src/stc/scintilla/lexlib -I../src/stc/scintilla/src -D__WX__ -DSCI_LEXER -DNO_CXX11_REGEX -DLINK_LEXERS -DwxUSE_BASE=1 -DWXMAKINGDLL -fPIC -DPIC -pthread -Wall -Wundef -Wunused-parameter -Wno-ctor-dtor-privacy -Woverloaded-virtual -O2 -I/data/data/com.termux/files/usr/include  -fvisibility=hidden -fvisibility-inlines-hidden   ../src/common/arcall.cpp
In file included from ../src/common/appbase.cpp:20:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:830:9: warning: 'HAVE_GCC_ATOMIC_BUILTINS' macro redefined [-Wmacro-redefined]
#define HAVE_GCC_ATOMIC_BUILTINS 1
        ^
../include/wx/android/config_android.h:27:9: note: previous definition is here
#define HAVE_GCC_ATOMIC_BUILTINS
        ^
In file included from ../src/common/appbase.cpp:20:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:978:9: warning: 'HAVE_NANOSLEEP' macro redefined [-Wmacro-redefined]
#define HAVE_NANOSLEEP 1
        ^
../include/wx/android/config_android.h:25:9: note: previous definition is here
#define HAVE_NANOSLEEP
        ^
In file included from ../src/common/appbase.cpp:20:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1080:9: warning: 'SIZEOF_VOID_P' macro redefined [-Wmacro-redefined]
#define SIZEOF_VOID_P 8
        ^
../include/wx/android/config_android.h:46:9: note: previous definition is here
#define SIZEOF_VOID_P 4
        ^
In file included from ../src/common/appbase.cpp:20:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1083:9: warning: 'SIZEOF_LONG' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 8
        ^
../include/wx/android/config_android.h:43:9: note: previous definition is here
#define SIZEOF_LONG 4
        ^
In file included from ../src/common/appbase.cpp:20:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1092:9: warning: 'SIZEOF_SIZE_T' macro redefined [-Wmacro-redefined]
#define SIZEOF_SIZE_T 8
        ^
../include/wx/android/config_android.h:45:9: note: previous definition is here
#define SIZEOF_SIZE_T 4
        ^
In file included from ../src/common/arcall.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:830:9: warning: 'HAVE_GCC_ATOMIC_BUILTINS' macro redefined [-Wmacro-redefined]
#define HAVE_GCC_ATOMIC_BUILTINS 1
        ^
../include/wx/android/config_android.h:27:9: note: previous definition is here
#define HAVE_GCC_ATOMIC_BUILTINS
        ^
In file included from ../src/common/arcall.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:978:9: warning: 'HAVE_NANOSLEEP' macro redefined [-Wmacro-redefined]
#define HAVE_NANOSLEEP 1
        ^
../include/wx/android/config_android.h:25:9: note: previous definition is here
#define HAVE_NANOSLEEP
        ^
In file included from ../src/common/arcall.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1080:9: warning: 'SIZEOF_VOID_P' macro redefined [-Wmacro-redefined]
#define SIZEOF_VOID_P 8
        ^
../include/wx/android/config_android.h:46:9: note: previous definition is here
#define SIZEOF_VOID_P 4
        ^
In file included from ../src/common/arcall.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1083:9: warning: 'SIZEOF_LONG' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 8
        ^
../include/wx/android/config_android.h:43:9: note: previous definition is here
#define SIZEOF_LONG 4
        ^
In file included from ../src/common/arcall.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1092:9: warning: 'SIZEOF_SIZE_T' macro redefined [-Wmacro-redefined]
#define SIZEOF_SIZE_T 8
        ^
../include/wx/android/config_android.h:45:9: note: previous definition is here
#define SIZEOF_SIZE_T 4
        ^
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps g++ -c -o monodll_arcfind.o       -I../src/regex -I../src/expat/expat/lib -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -D__WX__      -DWXBUILDING -I../src/stc/scintilla/include -I../src/stc/scintilla/lexlib -I../src/stc/scintilla/src -D__WX__ -DSCI_LEXER -DNO_CXX11_REGEX -DLINK_LEXERS -DwxUSE_BASE=1 -DWXMAKINGDLL -fPIC -DPIC -pthread -Wall -Wundef -Wunused-parameter -Wno-ctor-dtor-privacy -Woverloaded-virtual -O2 -I/data/data/com.termux/files/usr/include  -fvisibility=hidden -fvisibility-inlines-hidden   ../src/common/arcfind.cpp
5 warnings generated.
In file included from ../src/common/arcfind.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:830:9: warning: 'HAVE_GCC_ATOMIC_BUILTINS' macro redefined [-Wmacro-redefined]
#define HAVE_GCC_ATOMIC_BUILTINS 1
        ^
../include/wx/android/config_android.h:27:9: note: previous definition is here
#define HAVE_GCC_ATOMIC_BUILTINS
        ^
In file included from ../src/common/arcfind.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:978:9: warning: 'HAVE_NANOSLEEP' macro redefined [-Wmacro-redefined]
#define HAVE_NANOSLEEP 1
        ^
../include/wx/android/config_android.h:25:9: note: previous definition is here
#define HAVE_NANOSLEEP
        ^
In file included from ../src/common/arcfind.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1080:9: warning: 'SIZEOF_VOID_P' macro redefined [-Wmacro-redefined]
#define SIZEOF_VOID_P 8
        ^
../include/wx/android/config_android.h:46:9: note: previous definition is here
#define SIZEOF_VOID_P 4
        ^
In file included from ../src/common/arcfind.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1083:9: warning: 'SIZEOF_LONG' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 8
        ^
../include/wx/android/config_android.h:43:9: note: previous definition is here
#define SIZEOF_LONG 4
        ^
In file included from ../src/common/arcfind.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1092:9: warning: 'SIZEOF_SIZE_T' macro redefined [-Wmacro-redefined]
#define SIZEOF_SIZE_T 8
        ^
../include/wx/android/config_android.h:45:9: note: previous definition is here
#define SIZEOF_SIZE_T 4
        ^
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps g++ -c -o monodll_archive.o       -I../src/regex -I../src/expat/expat/lib -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -D__WX__      -DWXBUILDING -I../src/stc/scintilla/include -I../src/stc/scintilla/lexlib -I../src/stc/scintilla/src -D__WX__ -DSCI_LEXER -DNO_CXX11_REGEX -DLINK_LEXERS -DwxUSE_BASE=1 -DWXMAKINGDLL -fPIC -DPIC -pthread -Wall -Wundef -Wunused-parameter -Wno-ctor-dtor-privacy -Woverloaded-virtual -O2 -I/data/data/com.termux/files/usr/include  -fvisibility=hidden -fvisibility-inlines-hidden   ../src/common/archive.cpp
In file included from ../src/common/archive.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:830:9: warning: 'HAVE_GCC_ATOMIC_BUILTINS' macro redefined [-Wmacro-redefined]
#define HAVE_GCC_ATOMIC_BUILTINS 1
        ^
../include/wx/android/config_android.h:27:9: note: previous definition is here
#define HAVE_GCC_ATOMIC_BUILTINS
        ^
In file included from ../src/common/archive.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:978:9: warning: 'HAVE_NANOSLEEP' macro redefined [-Wmacro-redefined]
#define HAVE_NANOSLEEP 1
        ^
../include/wx/android/config_android.h:25:9: note: previous definition is here
#define HAVE_NANOSLEEP
        ^
In file included from ../src/common/archive.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1080:9: warning: 'SIZEOF_VOID_P' macro redefined [-Wmacro-redefined]
#define SIZEOF_VOID_P 8
        ^
../include/wx/android/config_android.h:46:9: note: previous definition is here
#define SIZEOF_VOID_P 4
        ^
In file included from ../src/common/archive.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1083:9: warning: 'SIZEOF_LONG' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 8
        ^
../include/wx/android/config_android.h:43:9: note: previous definition is here
#define SIZEOF_LONG 4
        ^
In file included from ../src/common/archive.cpp:10:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1092:9: warning: 'SIZEOF_SIZE_T' macro redefined [-Wmacro-redefined]
#define SIZEOF_SIZE_T 8
        ^
../include/wx/android/config_android.h:45:9: note: previous definition is here
#define SIZEOF_SIZE_T 4
        ^
In file included from ../src/common/arcall.cpp:16:
In file included from ../include/wx/zipstrm.h:16:
In file included from ../include/wx/archive.h:16:
In file included from ../include/wx/stream.h:19:
In file included from ../include/wx/object.h:19:
In file included from ../include/wx/memory.h:15:
../include/wx/string.h:561:7: warning: unknown warning group '-Wunsafe-loop-optimizations', ignored [-Wunknown-warning-option]
      wxGCC_ONLY_WARNING_SUPPRESS(unsafe-loop-optimizations)
      ^
../include/wx/defs.h:680:43: note: expanded from macro 'wxGCC_ONLY_WARNING_SUPPRESS'
#   define wxGCC_ONLY_WARNING_SUPPRESS(x) wxGCC_WARNING_SUPPRESS(x)
                                          ^
../include/wx/defs.h:668:9: note: expanded from macro 'wxGCC_WARNING_SUPPRESS'
        _Pragma (wxSTRINGIZE(GCC diagnostic ignored wxSTRINGIZE(wxCONCAT(-W,x))))
        ^
<scratch space>:15:25: note: expanded from here
 GCC diagnostic ignored "-Wunsafe-loop-optimizations"
                        ^
In file included from ../src/common/appbase.cpp:27:
In file included from ../include/wx/list.h:32:
In file included from ../include/wx/object.h:19:
In file included from ../include/wx/memory.h:15:
../include/wx/string.h:561:7: warning: unknown warning group '-Wunsafe-loop-optimizations', ignored [-Wunknown-warning-option]
      wxGCC_ONLY_WARNING_SUPPRESS(unsafe-loop-optimizations)
      ^
../include/wx/defs.h:680:43: note: expanded from macro 'wxGCC_ONLY_WARNING_SUPPRESS'
#   define wxGCC_ONLY_WARNING_SUPPRESS(x) wxGCC_WARNING_SUPPRESS(x)
                                          ^
../include/wx/defs.h:668:9: note: expanded from macro 'wxGCC_WARNING_SUPPRESS'
        _Pragma (wxSTRINGIZE(GCC diagnostic ignored wxSTRINGIZE(wxCONCAT(-W,x))))
        ^
<scratch space>:15:25: note: expanded from here
 GCC diagnostic ignored "-Wunsafe-loop-optimizations"
                        ^
In file included from ../src/common/arcfind.cpp:15:
In file included from ../include/wx/archive.h:16:
In file included from ../include/wx/stream.h:19:
In file included from ../include/wx/object.h:19:
In file included from ../include/wx/memory.h:15:
../include/wx/string.h:561:7: warning: unknown warning group '-Wunsafe-loop-optimizations', ignored [-Wunknown-warning-option]
      wxGCC_ONLY_WARNING_SUPPRESS(unsafe-loop-optimizations)
      ^
../include/wx/defs.h:680:43: note: expanded from macro 'wxGCC_ONLY_WARNING_SUPPRESS'
#   define wxGCC_ONLY_WARNING_SUPPRESS(x) wxGCC_WARNING_SUPPRESS(x)
                                          ^
../include/wx/defs.h:668:9: note: expanded from macro 'wxGCC_WARNING_SUPPRESS'
        _Pragma (wxSTRINGIZE(GCC diagnostic ignored wxSTRINGIZE(wxCONCAT(-W,x))))
        ^
<scratch space>:15:25: note: expanded from here
 GCC diagnostic ignored "-Wunsafe-loop-optimizations"
                        ^
In file included from ../src/common/any.cpp:15:
In file included from ../include/wx/any.h:19:
../include/wx/string.h:561:7: warning: unknown warning group '-Wunsafe-loop-optimizations', ignored [-Wunknown-warning-option]
      wxGCC_ONLY_WARNING_SUPPRESS(unsafe-loop-optimizations)
      ^
../include/wx/defs.h:680:43: note: expanded from macro 'wxGCC_ONLY_WARNING_SUPPRESS'
#   define wxGCC_ONLY_WARNING_SUPPRESS(x) wxGCC_WARNING_SUPPRESS(x)
                                          ^
../include/wx/defs.h:668:9: note: expanded from macro 'wxGCC_WARNING_SUPPRESS'
        _Pragma (wxSTRINGIZE(GCC diagnostic ignored wxSTRINGIZE(wxCONCAT(-W,x))))
        ^
<scratch space>:15:25: note: expanded from here
 GCC diagnostic ignored "-Wunsafe-loop-optimizations"
                        ^
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps g++ -c -o monodll_arrstr.o       -I../src/regex -I../src/expat/expat/lib -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -D__WX__      -DWXBUILDING -I../src/stc/scintilla/include -I../src/stc/scintilla/lexlib -I../src/stc/scintilla/src -D__WX__ -DSCI_LEXER -DNO_CXX11_REGEX -DLINK_LEXERS -DwxUSE_BASE=1 -DWXMAKINGDLL -fPIC -DPIC -pthread -Wall -Wundef -Wunused-parameter -Wno-ctor-dtor-privacy -Woverloaded-virtual -O2 -I/data/data/com.termux/files/usr/include  -fvisibility=hidden -fvisibility-inlines-hidden   ../src/common/arrstr.cpp
In file included from ../src/common/arrstr.cpp:16:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:830:9: warning: 'HAVE_GCC_ATOMIC_BUILTINS' macro redefined [-Wmacro-redefined]
#define HAVE_GCC_ATOMIC_BUILTINS 1
        ^
../include/wx/android/config_android.h:27:9: note: previous definition is here
#define HAVE_GCC_ATOMIC_BUILTINS
        ^
In file included from ../src/common/arrstr.cpp:16:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:978:9: warning: 'HAVE_NANOSLEEP' macro redefined [-Wmacro-redefined]
#define HAVE_NANOSLEEP 1
        ^
../include/wx/android/config_android.h:25:9: note: previous definition is here
#define HAVE_NANOSLEEP
        ^
In file included from ../src/common/arrstr.cpp:16:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1080:9: warning: 'SIZEOF_VOID_P' macro redefined [-Wmacro-redefined]
#define SIZEOF_VOID_P 8
        ^
../include/wx/android/config_android.h:46:9: note: previous definition is here
#define SIZEOF_VOID_P 4
        ^
In file included from ../src/common/arrstr.cpp:16:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1083:9: warning: 'SIZEOF_LONG' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 8
        ^
../include/wx/android/config_android.h:43:9: note: previous definition is here
#define SIZEOF_LONG 4
        ^
In file included from ../src/common/arrstr.cpp:16:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1092:9: warning: 'SIZEOF_SIZE_T' macro redefined [-Wmacro-redefined]
#define SIZEOF_SIZE_T 8
        ^
../include/wx/android/config_android.h:45:9: note: previous definition is here
#define SIZEOF_SIZE_T 4
        ^
In file included from ../src/common/arcall.cpp:16:
In file included from ../include/wx/zipstrm.h:16:
In file included from ../include/wx/archive.h:17:
../include/wx/filename.h:468:43: error: use of overloaded operator '[]' is ambiguous (with operand types 'wxString' and 'unsigned int')
        { return GetPathSeparators(format)[0u]; }
                 ~~~~~~~~~~~~~~~~~~~~~~~~~^~~
../include/wx/string.h:1525:18: note: candidate function
    wxUniCharRef operator[](int n)
                 ^
../include/wx/string.h:1527:18: note: candidate function
    wxUniCharRef operator[](long n)
                 ^
../include/wx/string.h:1529:18: note: candidate function
    wxUniCharRef operator[](size_t n)
                 ^
In file included from ../src/common/archive.cpp:15:
In file included from ../include/wx/archive.h:16:
In file included from ../include/wx/stream.h:19:
In file included from ../include/wx/object.h:19:
In file included from ../include/wx/memory.h:15:
../include/wx/string.h:561:7: warning: unknown warning group '-Wunsafe-loop-optimizations', ignored [-Wunknown-warning-option]
      wxGCC_ONLY_WARNING_SUPPRESS(unsafe-loop-optimizations)
      ^
../include/wx/defs.h:680:43: note: expanded from macro 'wxGCC_ONLY_WARNING_SUPPRESS'
#   define wxGCC_ONLY_WARNING_SUPPRESS(x) wxGCC_WARNING_SUPPRESS(x)
                                          ^
../include/wx/defs.h:668:9: note: expanded from macro 'wxGCC_WARNING_SUPPRESS'
        _Pragma (wxSTRINGIZE(GCC diagnostic ignored wxSTRINGIZE(wxCONCAT(-W,x))))
        ^
<scratch space>:15:25: note: expanded from here
 GCC diagnostic ignored "-Wunsafe-loop-optimizations"
                        ^
In file included from ../src/common/arcfind.cpp:15:
In file included from ../include/wx/archive.h:17:
../include/wx/filename.h:468:43: error: use of overloaded operator '[]' is ambiguous (with operand types 'wxString' and 'unsigned int')
        { return GetPathSeparators(format)[0u]; }
                 ~~~~~~~~~~~~~~~~~~~~~~~~~^~~
../include/wx/string.h:1525:18: note: candidate function
    wxUniCharRef operator[](int n)
                 ^
../include/wx/string.h:1527:18: note: candidate function
    wxUniCharRef operator[](long n)
                 ^
../include/wx/string.h:1529:18: note: candidate function
    wxUniCharRef operator[](size_t n)
                 ^
In file included from ../src/common/archive.cpp:15:
In file included from ../include/wx/archive.h:17:
../include/wx/filename.h:468:43: error: use of overloaded operator '[]' is ambiguous (with operand types 'wxString' and 'unsigned int')
        { return GetPathSeparators(format)[0u]; }
                 ~~~~~~~~~~~~~~~~~~~~~~~~~^~~
../include/wx/string.h:1525:18: note: candidate function
    wxUniCharRef operator[](int n)
                 ^
../include/wx/string.h:1527:18: note: candidate function
    wxUniCharRef operator[](long n)
                 ^
../include/wx/string.h:1529:18: note: candidate function
    wxUniCharRef operator[](size_t n)
                 ^
5 warnings generated.
/data/data/com.termux/files/home/wxWidgets/buildgtk/bk-deps g++ -c -o monodll_base64.o       -I../src/regex -I../src/expat/expat/lib -I/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1 -I../include -D_FILE_OFFSET_BITS=64  -D__WX__      -DWXBUILDING -I../src/stc/scintilla/include -I../src/stc/scintilla/lexlib -I../src/stc/scintilla/src -D__WX__ -DSCI_LEXER -DNO_CXX11_REGEX -DLINK_LEXERS -DwxUSE_BASE=1 -DWXMAKINGDLL -fPIC -DPIC -pthread -Wall -Wundef -Wunused-parameter -Wno-ctor-dtor-privacy -Woverloaded-virtual -O2 -I/data/data/com.termux/files/usr/include  -fvisibility=hidden -fvisibility-inlines-hidden   ../src/common/base64.cpp
In file included from ../src/common/arrstr.cpp:19:
In file included from ../include/wx/arrstr.h:15:
../include/wx/string.h:561:7: warning: unknown warning group '-Wunsafe-loop-optimizations', ignored [-Wunknown-warning-option]
      wxGCC_ONLY_WARNING_SUPPRESS(unsafe-loop-optimizations)
      ^
../include/wx/defs.h:680:43: note: expanded from macro 'wxGCC_ONLY_WARNING_SUPPRESS'
#   define wxGCC_ONLY_WARNING_SUPPRESS(x) wxGCC_WARNING_SUPPRESS(x)
                                          ^
../include/wx/defs.h:668:9: note: expanded from macro 'wxGCC_WARNING_SUPPRESS'
        _Pragma (wxSTRINGIZE(GCC diagnostic ignored wxSTRINGIZE(wxCONCAT(-W,x))))
        ^
<scratch space>:15:25: note: expanded from here
 GCC diagnostic ignored "-Wunsafe-loop-optimizations"
                        ^
6 warnings and 1 error generated.
In file included from ../src/common/base64.cpp:9:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:830:9: warning: 'HAVE_GCC_ATOMIC_BUILTINS' macro redefined [-Wmacro-redefined]
#define HAVE_GCC_ATOMIC_BUILTINS 1
        ^
../include/wx/android/config_android.h:27:9: note: previous definition is here
#define HAVE_GCC_ATOMIC_BUILTINS
        ^
In file included from ../src/common/base64.cpp:9:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:978:9: warning: 'HAVE_NANOSLEEP' macro redefined [-Wmacro-redefined]
#define HAVE_NANOSLEEP 1
        ^
../include/wx/android/config_android.h:25:9: note: previous definition is here
#define HAVE_NANOSLEEP
        ^
In file included from ../src/common/base64.cpp:9:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1080:9: warning: 'SIZEOF_VOID_P' macro redefined [-Wmacro-redefined]
#define SIZEOF_VOID_P 8
        ^
../include/wx/android/config_android.h:46:9: note: previous definition is here
#define SIZEOF_VOID_P 4
        ^
In file included from ../src/common/base64.cpp:9:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1083:9: warning: 'SIZEOF_LONG' macro redefined [-Wmacro-redefined]
#define SIZEOF_LONG 8
        ^
../include/wx/android/config_android.h:43:9: note: previous definition is here
#define SIZEOF_LONG 4
        ^
In file included from ../src/common/base64.cpp:9:
In file included from ../include/wx/wxprec.h:12:
In file included from ../include/wx/defs.h:45:
In file included from ../include/wx/platform.h:159:
/data/data/com.termux/files/home/wxWidgets/buildgtk/lib/wx/include/base-unicode-3.1/wx/setup.h:1092:9: warning: 'SIZEOF_SIZE_T' macro redefined [-Wmacro-redefined]
#define SIZEOF_SIZE_T 8
        ^
../include/wx/android/config_android.h:45:9: note: previous definition is here
#define SIZEOF_SIZE_T 4
        ^
make: *** [Makefile:15637: monodll_arcall.o] Error 1
make: *** Waiting for unfinished jobs....
6 warnings and 1 error generated.
6 warnings and 1 error generated.
make: *** [Makefile:15640: monodll_arcfind.o] Error 1
make: *** [Makefile:15643: monodll_archive.o] Error 1
In file included from ../src/common/appbase.cpp:39:
../include/wx/filename.h:468:43: error: use of overloaded operator '[]' is ambiguous (with operand types 'wxString' and 'unsigned int')
        { return GetPathSeparators(format)[0u]; }
                 ~~~~~~~~~~~~~~~~~~~~~~~~~^~~
../include/wx/string.h:1525:18: note: candidate function
    wxUniCharRef operator[](int n)
                 ^
../include/wx/string.h:1527:18: note: candidate function
    wxUniCharRef operator[](long n)
                 ^
../include/wx/string.h:1529:18: note: candidate function
    wxUniCharRef operator[](size_t n)
                 ^
In file included from ../src/common/base64.cpp:14:
In file included from ../include/wx/base64.h:16:
../include/wx/string.h:561:7: warning: unknown warning group '-Wunsafe-loop-optimizations', ignored [-Wunknown-warning-option]
      wxGCC_ONLY_WARNING_SUPPRESS(unsafe-loop-optimizations)
      ^
../include/wx/defs.h:680:43: note: expanded from macro 'wxGCC_ONLY_WARNING_SUPPRESS'
#   define wxGCC_ONLY_WARNING_SUPPRESS(x) wxGCC_WARNING_SUPPRESS(x)
                                          ^
../include/wx/defs.h:668:9: note: expanded from macro 'wxGCC_WARNING_SUPPRESS'
        _Pragma (wxSTRINGIZE(GCC diagnostic ignored wxSTRINGIZE(wxCONCAT(-W,x))))
        ^
<scratch space>:15:25: note: expanded from here
 GCC diagnostic ignored "-Wunsafe-loop-optimizations"
                        ^
6 warnings and 1 error generated.
make: *** [Makefile:15634: monodll_appbase.o] Error 1
6 warnings generated.
6 warnings generated.
6 warnings generated.
```
make install (remaining due to error in previous step)


3) 
```
git clone https://github.com/alexkay/spek 
cd spek*2
autoreconf -vf
./autogen.sh
```
cat log
```
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: running: intltoolize --automake --copy --force
autoreconf: running: aclocal --force 
configure.ac:40: warning: macro 'AM_OPTIONS_WXCONFIG' not found in library
configure.ac:42: warning: macro 'AM_PATH_WXCONFIG' not found in library
autoreconf: configure.ac: tracing
configure.ac: warning: AM_GNU_GETTEXT_VERSION or AM_GNU_GETTEXT_REQUIRE_VERSION is used, but not AM_GNU_GETTEXT
autoreconf: configure.ac: not using Libtool
autoreconf: running: intltoolize --copy --force
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: /data/data/com.termux/files/usr/bin/autoconf --force
configure.ac:40: error: possibly undefined macro: AM_OPTIONS_WXCONFIG
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.ac:42: error: possibly undefined macro: AM_PATH_WXCONFIG
autoreconf: error: /data/data/com.termux/files/usr/bin/autoconf failed with exit status: 1

```
```
configure --prefix=$PREFIX --build=aarch64-unknown-linux-gnu (step remaining)
```

--- a/packages/spek/build.sh
+++ b/packages/spek/build.sh
@@ -1,0 +1,33 @@
+TERMUX_PKG_HOMEPAGE=http://spek.cc/
+TERMUX_PKG_DESCRIPTION="An acoustic spectrum analyser"
+TERMUX_PKG_LICENSE="GPL-3.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=0.8.3_p20181228
+_COMMIT=f071c2956176ad53c7c8059e5c00e694ded31ded
+TERMUX_PKG_SRCURL=https://github.com/alexkay/spek.git
+TERMUX_PKG_GIT_BRANCH=master
+TERMUX_PKG_DEPENDS="ffmpeg, wxwidgets"
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="ac_cv_path_WX_CONFIG_PATH=$TERMUX_PREFIX/bin/wx-config"
+
+termux_step_post_get_source() {
+	git fetch --unshallow
+	git checkout $_COMMIT
+}
+
+termux_step_pre_configure() {
+	AUTOPOINT='intltoolize --automake --copy' \
+		autoreconf -fi --include=$TERMUX_PREFIX/share/aclocal
+
+	_NEED_DUMMY_LIBPTHREAD_A=
+	_LIBPTHREAD_A=$TERMUX_PREFIX/lib/libpthread.a
+	if [ ! -e $_LIBPTHREAD_A ]; then
+		_NEED_DUMMY_LIBPTHREAD_A=true
+		echo '!<arch>' > $_LIBPTHREAD_A
+	fi
+}
+
+termux_step_post_make_install() {
+	if [ $_NEED_DUMMY_LIBPTHREAD_A ]; then
+		rm -f $_LIBPTHREAD_A
+	fi
+}

--- a/packages/spek/src-spek-preferences.cc.patch
+++ b/packages/spek/src-spek-preferences.cc.patch
@@ -1,0 +1,25 @@
+--- a/src/spek-preferences.cc
++++ b/src/spek-preferences.cc
+@@ -12,6 +12,7 @@
+ 
+ void SpekPreferences::init()
+ {
++#if wxUSE_INTL
+     if (this->locale) {
+         delete this->locale;
+     }
+@@ -27,9 +28,14 @@
+     }
+     this->locale->Init(lang);
+     this->locale->AddCatalog(GETTEXT_PACKAGE);
++#endif
+ }
+ 
++#if wxUSE_INTL
+ SpekPreferences::SpekPreferences() : locale(NULL)
++#else
++SpekPreferences::SpekPreferences()
++#endif
+ {
+     wxString path = spek_platform_config_path("spek");
+     this->config = new wxFileConfig(

--- a/packages/spek/src-spek-preferences.h.patch
+++ b/packages/spek/src-spek-preferences.h.patch
@@ -1,0 +1,11 @@
+--- a/src/spek-preferences.h
++++ b/src/spek-preferences.h
+@@ -21,6 +21,8 @@
+     SpekPreferences(const SpekPreferences&);
+     void operator=(const SpekPreferences&);
+ 
++#if wxUSE_INTL
+     wxLocale *locale;
++#endif
+     wxFileConfig *config;
+ };

